### PR TITLE
Fix inconsistent timeout behaviour of check_ping plugin: Alternative 2

### PR DESCRIPTION
**Pull request by bazzisoft**
This PR is an alternative to https://github.com/nagios-plugins/nagios-plugins/pull/502 and attempts to resolve the same issue, while ensuring `check_ping` does not send more than `-c max_packets` packets for dead hosts.

For unresponsive hosts instead of the ping command timing out and providing useful output, viz:
`PING CRITICAL - Packet loss = 100%|rta=1200.000000ms;600.000000`
we get instead:
`CRITICAL - Plugin timed out`

The PR essentially includes 2 separate fixes:

1. The timeout parameters for `ping` (`-t`) and `ping6` (`-X`) used on **FreeBSD** are not compatible with the timeout parameter now used in linux (`-W`). The former are global timeouts whereas the latter is the maximum wait time for the _last packet sent_. So, the former were converted to `-W` and `-x` respectively per these docs:
    - https://www.freebsd.org/cgi/man.cgi?query=ping&apropos=0&sektion=8&manpath=FreeBSD+10.2-RELEASE&arch=default&format=html
    - https://www.freebsd.org/cgi/man.cgi?query=ping6&apropos=0&sektion=8&manpath=FreeBSD+10.2-RELEASE&arch=default&format=html

2. Since the the linux "deadline" parameter (`-w`) was changed to the "timeout" (`-W`) (see https://github.com/nagios-plugins/nagios-plugins/issues/139), the ping commands execute for longer than the SIGALRM defined for the plugin. This is because the `-W` param is effectively a timeout for the last ping packet sent, so the ping command is executing for a max "timeout + num packets" seconds instead of "timeout" seconds. Thus the SIGALRM set to "timeout + 1" seconds is triggering and killing the plugin.

    The proposed fix is to set the timeout passed to the `ping` command to: 
    `MAX(1, timeout - num_packets)`

Tested on Ubuntu 18.04 and FreeBSD 11.3.

### CODE REVIEW

**Comment by bazzisoft** [comment]
@sawolf Please also consider this PR as an alternative to https://github.com/nagios-plugins/nagios-plugins/pull/502 which we discussed yesterday; I'm not really sure which is preferred.

**Comment by bazzisoft** [comment]
@sawolf For what it's worth, in our extended (albeit not exactly scientific) testing this alternative seems a little more reliable & consistent between platforms than the previous https://github.com/nagios-plugins/nagios-plugins/pull/502. We'll be using this approach in our fork meantime.

**Comment by sawolf** [comment]
Accepted #502 - closing this.
